sdk/instrumentation/net/http/handler.go

```diff
@@ -87,12 +88,12 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 		// Only records the body if it is not empty and the content type
 		// header is not streamable
-		if shouldRecordBody && len(body) > 0 {
+		if len(body) > 0 {
 			span.SetAttribute("http.request.body", string(body))
 		}
 
 		// run body filters
-		if h.requestFilter.EvaluateBody(span, body) {
+		if h.filter.EvaluateBody(span, body) {
```

Comment:
Notice this filter is only applied if the dataCapture for body is enabled which is IMO an ortogonal concern to recording of body.

Reply:
See comment here: https://github.com/hypertrace/goagent/pull/83#discussion_r578867109.